crates/holochain_state/src/mutations.rs

```diff
@@ -0,0 +1,513 @@
+use crate::agent_info::AgentKvKey;
+use crate::entry_def::EntryDefStoreKey;
+use crate::prelude::SignedValidationReceipt;
+use crate::query::to_blob;
+use crate::scratch::Scratch;
+use crate::validation_db::ValidationLimboStatus;
+use holo_hash::encode::blake2b_256;
+use holo_hash::*;
+use holochain_sqlite::rusqlite::named_params;
+use holochain_sqlite::rusqlite::Transaction;
+use holochain_types::dht_op::DhtOpHashed;
+use holochain_types::dht_op::DhtOpLight;
+use holochain_types::dht_op::OpOrder;
+use holochain_types::prelude::DhtOpError;
+use holochain_types::prelude::DnaDefHashed;
+use holochain_types::prelude::DnaWasmHashed;
+use holochain_types::EntryHashed;
+use holochain_zome_types::*;
+
+pub use error::*;
+
+mod error;
+
+#[macro_export]
+macro_rules! sql_insert {
+    ($txn:expr, $table:ident, { $($field:literal : $val:expr , )+ $(,)? }) => {{
+        let table = stringify!($table);
+        let fieldnames = &[ $( { $field } ,)+ ].join(",");
+        let fieldvars = &[ $( { format!(":{}", $field) } ,)+ ].join(",");
+        let sql = format!("INSERT INTO {} ({}) VALUES ({})", table, fieldnames, fieldvars);
+        $txn.execute_named(&sql, &[$(
+            (format!(":{}", $field).as_str(), &$val as &dyn holochain_sqlite::rusqlite::ToSql),
+        )+])
+    }};
+}
+
+macro_rules! dht_op_update {
+    ($txn:expr, $hash:expr, { $($field:literal : $val:expr , )+ $(,)? }) => {{
+        let fieldvars = &[ $( { format!("{} = :{}", $field, $field) } ,)+ ].join(",");
+        let sql = format!(
+            "
+            UPDATE DhtOp 
+            SET {}
+            WHERE DhtOp.hash = :hash
+            ", fieldvars);
+        $txn.execute(&sql, &[
+            (":hash", &$hash as &dyn holochain_sqlite::rusqlite::ToSql),
+            $(
+            (format!(":{}", $field).as_str(), &$val as &dyn holochain_sqlite::rusqlite::ToSql),
+        )+])
+    }};
+}
+
+macro_rules! dht_op_add_one {
+    ($txn:expr, $hash:expr, $field:literal) => {{
+        let fieldvars = format!("{} = IFNULL({}, 0) + 1", $field, $field);
+        let sql = format!(
+            "
+            UPDATE DhtOp 
+            SET {}
+            WHERE DhtOp.hash = :hash
+            ",
+            fieldvars
+        );
+        $txn.execute(
+            &sql,
+            &[(":hash", &$hash as &dyn holochain_sqlite::rusqlite::ToSql)],
+        )
```

Comment:
I don't think this macro adds clarity, or saves boilerplate... and it's adding runtime string manipulation.
the below `add_one_receipt_count` function could be changed to:

```rust
pub fn add_one_receipt_count(txn: &mut Transaction, hash: &DhtOpHash) -> StateMutationResult<()> {
    txn.execute(
        "UPDATE DhtOp SET receipt_count = IFNULL(receipt_count, 0) + 1 WHERE hash = :hash;",
        named_params! { ":hash" = hash },
    )?;
    Ok(())
}
```